common/stacks/src/main/scala/weco/api/stacks/services/SierraService.scala

```diff
@@ -235,44 +230,22 @@ class SierraService(
     }
   }
 
-  protected def buildStacksHold(entry: SierraHold): StacksHold = {
-    val itemNumber = SierraItemIdentifier.fromUrl(entry.record)
-    val sourceIdentifier = SierraItemIdentifier.toSourceIdentifier(itemNumber)
-
-    val pickupLocation = StacksPickupLocation(
-      id = entry.pickupLocation.code,
-      label = entry.pickupLocation.name
-    )
-
-    val pickup = StacksPickup(
-      location = pickupLocation,
-      pickUpBy = entry.pickupByDate
-    )
-
-    val status = StacksHoldStatus(
-      id = entry.status.code,
-      label = entry.status.name
-    )
-
-    StacksHold(sourceIdentifier, pickup, status)
-  }
-
-  def getStacksUserHolds(
-    patronNumber: SierraPatronNumber
-  ): Future[Either[SierraErrorCode, StacksUserHolds]] =
-    sierraSource
-      .listHolds(patronNumber)
-      .map {
-        case Right(holds) =>
-          Right(
-            StacksUserHolds(
-              userId = patronNumber.withoutCheckDigit,
-              holds = holds.entries.map(buildStacksHold)
-            )
-          )
+  def getHolds(patronNumber: SierraPatronNumber) =
+    for {
+      holds <- sierraSource
+        .listHolds(patronNumber)
+        .flatMap {
+          case Right(holds) => Future(holds)
```

Comment:
```suggestion
          case Right(holds) => Future.successful(holds)
```

Let's be explicit and match the left arm.